### PR TITLE
tests: Disable K8s upstream tests that we do not support

### LINKED ISSUE #17796: k8s-upstream: SCTP tests failing
The following tests are consistently failing in [the k8s-upstream CI job](https://jenkins.cilium.io/job/cilium-master-k8s-upstream/):
```
[Fail] [sig-network] Services [It] should respect internalTrafficPolicy=Local Pod to Pod (hostNetwork: true) [Feature:ServiceInternalTrafficPolicy] 
/home/vagrant/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/service.go:2267

[Fail] [sig-network] Services [It] should respect internalTrafficPolicy=Local Pod (hostNetwork: true) to Pod (hostNetwork: true) [Feature:ServiceInternalTrafficPolicy] 
/home/vagrant/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/service.go:2349

[Fail] [sig-network] Services [It] should respect internalTrafficPolicy=Local Pod (hostNetwork: true) to Pod [Feature:ServiceInternalTrafficPolicy] 
/home/vagrant/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/service.go:2187

[Fail] [sig-network] Netpol NetworkPolicy between server and client [It] should properly isolate pods that are selected by a policy allowing SCTP, even if the plugin doesn't support SCTP [Feature:NetworkPolicy] 
/home/vagrant/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/netpol/network_policy.go:1096
```

### PR DESCRIPTION
**Pull request by nathanjsweet**
Tests for internalTrafficPolicy=Local and SCTP that were added
in 1.22 should be disabled as we support neither scenario

Signed-off-by: Nate Sweet <nathanjsweet@pm.me>

Please ensure your pull request adheres to the following guidelines:

Fixes: #17796

### CODE REVIEW

File: test/kubernetes-test.sh [hunk 1]
```diff
@@ -102,4 +102,4 @@ ${HOME}/go/bin/kubetest --provider=local --test \
 # should enforce ingress policy allowing any port traffic to a server on a specific protocol
```
Comment (aditighag) [inline test/kubernetes-test.sh:102]: Please add - 
"Cilium does not support internalTrafficPolicy as Local Redirect Policy allows user to enable node-local redirection."

**Review by aditighag — CHANGES_REQUESTED** [review]

**Review by pchaigno — APPROVED** [review]
Pending Aditi's review.

**Review by aditighag — APPROVED** [review]

**Comment by pchaigno** [comment]
test-upstream-k8s

File: test/kubernetes-test.sh [hunk 2]
```diff
@@ -100,6 +100,10 @@ ${HOME}/go/bin/kubetest --provider=local --test \
...
+  --test_args="--ginkgo.focus=Net.*ol.* --e2e-verify-service-account=false --host ${KUBE_MASTER_URL} --ginkgo.skip=(should.not.allow.access.by.TCP.when.a.policy.specifies.only.SCTP)|(should.allow.egress.access.to.server.in.CIDR.block)|(should.enforce.except.clause.while.egress.access.to.server.in.CIDR.block)|(should.ensure.an.IP.overlapping.both.IPBlock.CIDR.and.IPBlock.Except.is.allowed)|(NetworkPolicy.between.server.and.client.using.SCTP)|(should.enforce.ingress.policy.allowing.any.port.traffic.to.a.server.on.a.specific.protocol)|(should.respect.internalTrafficPolicy.Local.Pod.to.Pod..Feature.ServiceInternalTrafficPolicy.)|(should.properly.isolate.pods.that.are.selected.by.a.policy.allowing.SCTP..even.if.the.plugin.doesn.t.support.SCTP..Feature.NetworkPolicy.)"
```
Comment (nbusseneau) [inline test/kubernetes-test.sh]: Suggestion:
```suggestion
  --test_args="--ginkgo.focus=Net.*ol.* --e2e-verify-service-account=false --host ${KUBE_MASTER_URL} --ginkgo.skip=(should.not.allow.access.by.TCP.when.a.policy.specifies.only.SCTP)|(should.allow.egress.access.to.server.in.CIDR.block)|(should.enforce.except.clause.while.egress.access.to.server.in.CIDR.block)|(should.ensure.an.IP.overlapping.both.IPBlock.CIDR.and.IPBlock.Except.is.allowed)|(NetworkPolicy.between.server.and.client.using.SCTP)|(should.enforce.ingress.policy.allowing.any.port.traffic.to.a.server.on.a.specific.protocol)|(should.respect.internalTrafficPolicy.Local.Pod.*to.Pod.*.Feature.ServiceInternalTrafficPolicy.)|(should.properly.isolate.pods.that.are.selected.by.a.policy.allowing.SCTP..even.if.the.plugin.doesn.t.support.SCTP..Feature.NetworkPolicy.)"
```
To cover remaining CI failures:
```
should respect internalTrafficPolicy=Local Pod (hostNetwork: true) to Pod [Feature:ServiceInternalTrafficPolicy] 
should respect internalTrafficPolicy=Local Pod (hostNetwork: true) to Pod (hostNetwork: true) [Feature:ServiceInternalTrafficPolicy] 
should respect internalTrafficPolicy=Local Pod to Pod (hostNetwork: true) [Feature:ServiceInternalTrafficPolicy] 
```

**Review by nbusseneau — CHANGES_REQUESTED** [review]

**Review by nbusseneau — APPROVED** [review]
Upstream test passed. This PR does not need to run any additional CI and reviews are in, marking as ready to merge.

**Pull request merged**